app/assets/javascripts/sufia/select_work_type.es6

```diff
@@ -27,7 +27,9 @@ export default class SelectWorkType {
 
   // when the form is submitted route to the correct location
   routingLogic(e) {
-      e.preventDefault();
+      e.preventDefault()
+      if (this.destination() === undefined)
```

Comment:
This is the only code added here: check the destination and short-circuit (by returning `false`) if undefined.